firebase-firestore/src/main/java/com/google/firebase/firestore/local/BundleCache.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import androidx.annotation.Nullable;
+
+/** Provides methods to save and read Firestore bundles. */
+public interface BundleCache {
+  /**
+   * Gets the saved BundleMetadata for a given bundle id. Returns {@code null} if no bundles are
```

Comment:
In general, I tend to use literal references for types that are used in the signature that I am commenting on. I think you can reason about it if you replace `code` with `link` (which I would do if I implemented your suggestion). This would look as such:

```
/**
   * Gets the saved {@link BundleMetadata} for a given bundle id.
   */
BundleMetadata getBundleMetadata(String bundleId);
```

The generated JavaDoc HTML would not how two links to BundleMetadata, as the return type already links to the source location.

Let me know if this is convincing. If not, I'll change it :)